core/templates/dev/head/components/hint_and_solution_buttons_directive.html

```diff
@@ -0,0 +1,27 @@
+<div ng-if="isWindowNarrow && isHintButtonVisibleAtIndex(0)" class="oppia-hint-and-solution-buttons pull-left">
+  <span class="pull-left footer-hints-label">HINTS</span>
+  <ul class="list-inline pull-right">
+    <li ng-repeat="hintPosition in hintPositions">
+      <img class="oppia-hint-button" ng-if="isHintButtonVisibleAtIndex(hintPosition)" ng-src="<[activeModalIndex === hintPosition ? activeHintUrl : defaultHintUrl]>" ng-click="displayHintModalForIndex(hintPosition)">
+    </li>
+    <li ng-if="isSolutionButtonVisible()">
+      <img class="oppia-solution-button" ng-src="<[activeModalIndex === -1 ? activeSolutionUrl : defaultSolutionUrl]>" ng-click="displaySolutionModal()">
```

Comment:
Hmm I feel a bit weird having this 'activeModalIndex' arbitrarily get set to -1 for solution, as it's not really an index. Could you just use a separate boolean such as $scope.solutionModalIsActive or similar to handle solutions?